guide/guide-qip.rst

```diff
@@ -0,0 +1,264 @@
+
+.. code-block:: python
+
+      >>> from qutip.qip.device import OptPulseProcessor
+      >>> from qutip.qip import QubitCircuit
+      >>> from qutip import tensor, sigmaz, sigmax, sigmay
+      >>> # Same parameter for all the gates
+      ... qc = QubitCircuit(N=1)
+      >>> qc.add_gate("SNOT", 0)
+      >>>
+      >>> num_tslots = 10
+      >>> evo_time = 10
+      >>> processor = OptPulseProcessor(N=1, drift=sigmaz(), ctrls=[sigmax()])
+      >>> # num_tslots and evo_time are two keyword arguments
+      ... tlist, coeffs = processor.load_circuit(
+      ... qc, num_tslots=num_tslots, evo_time=evo_time)
+      >>>
+      >>> # Different parameters for different gates
+      ... qc = QubitCircuit(N=2)
+      >>> qc.add_gate("SNOT", 0)
+      >>> qc.add_gate("SWAP", targets=[0, 1])
+      >>> qc.add_gate('CNOT', controls=1, targets=[0])
+      >>>
+      >>> processor = OptPulseProcessor(N=2, drift=tensor([sigmaz()]*2))
+      >>> processor.add_ctrl(sigmax(), cyclic_permutation=True)
+      >>> processor.add_ctrl(sigmay(), cyclic_permutation=True)
+      >>> processor.add_ctrl(tensor([sigmay(), sigmay()]))
+      >>> setting_args = {"SNOT": {"num_tslots": 10, "evo_time": 1},
+      ...             "SWAP": {"num_tslots": 30, "evo_time": 3},
+      ...             "CNOT": {"num_tslots": 30, "evo_time": 3}}
+      >>> tlist, coeffs = processor.load_circuit(
+      ... qc, setting_args=setting_args, merge_gates=False)
+
+Noise Simulation
+================
+
+In the common way of QIP simulation, where evolution is carried out by gate matrix production, the noise is usually simulated with bit flipping and sign flipping errors. The typical approaches are either applying bit/sign flipping gate probabilistically or applying Kraus operators representing different noisy channels (e.g. amplitude damping, dephasing) after each unitary gate evolution. Those two ways are equivalent and the parameters in the Kraus operators are exactly the probability of a flipping error happens during the gate operation time.
+
+Since the circuit processor simulates the state evolution at the level of driving Hamiltonian, there is no way to apply an error operator to the continuous time evolution. Instead, the error is added to the driving Hamiltonian list or the collapse operators contributing to the evolution. Mathematically, this is no different from adding flipping errors probabilistically (It is actually how :func:`qutip.mcsolve` works internally). The collapse operator for amplitude damping and dephasing are exactly the destroying operator and the sign-flipping operator. One just needs to choose the correct coefficients for them to simulate the noise, e,g, the relaxation time T1, T2. Because it is based on the open system evolution instead of abstract operators, this simulation is closer to the physical implementation and also more general.
```

Comment:
* the driving Hamiltonian
* it is actually how (lower case "it")
* e,g, ---> e.g.
* relaxation time, T1, dephasing time, T2